### PR TITLE
feature/rework_navbar

### PR DESCRIPTION
**Pull request by dephiloper**
implemented fully working bottom bar navigation w/ seperate navigation stack for each segment

For testing purpose I created under the segment 'More' two buttons to show how the navigation to subpages is done (Help and Credits Page). Additionally when pressing the '?'-Button the app navigates to Help where two testing buttons are implemented. These explain the navigation to a different segment from an subpage, with and without popping the current segments subpages.

### CODE REVIEW

File: lib/app.dart [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import 'package:flutter/material.dart';
+import 'package:hiking4nerds/navigation/segment_navigator.dart';
+import 'navigation/bottom_navigation.dart';
+
+class App extends StatefulWidget {
```
Comment (jakubprogramming) [inline lib/app.dart:5]: new file for our App. Much better than having everything inside the Main. Great!

File: lib/pages/home.dart [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import 'package:flutter/material.dart';
 import 'package:hiking4nerds/components/mapwidget.dart';
-import 'package:hiking4nerds/components/navbar.dart';
 import 'package:hiking4nerds/components/shareroute.dart';
 import 'package:hiking4nerds/services/osmdata.dart';
 import 'package:hiking4nerds/services/route.dart';
 import 'package:hiking4nerds/styles.dart';
 
+// TODO rename to map
```
Comment (jakubprogramming) [inline lib/pages/home.dart:8]: if we want to rename it to map, why not now?

**Review by jakubprogramming — COMMENTED** [review]
Looks good for the most part!

After merging develop into this branch I had to fix a range of merge conflicts.
I also added some additional code.

Furthermore (you probably didn't notice) the marker in the location_selection was not in the correct position anymore. This is due to the fact that the navbar changed. I fixed that error.

Since I did some changes myself its probably best somebody approves these changes before we merge this PR.

**Review by jakubprogramming — APPROVED** [review]
After I have merged develop into this branch and fixing merge conflicts and some minor errors this gets my approval. 

Now somebody may want to approve my changes before merging.

File: lib/navigation/bottom_navigation.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:hiking4nerds/styles.dart';
+
```
Comment (Capscreen) [inline lib/navigation/bottom_navigation.dart:3]: clean and elegant solution for navigation, looks really good.

**Review by Capscreen — APPROVED** [review]
great solution so far.

**Pull request merged**